src/server.rs

```diff
@@ -0,0 +1,145 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::Mutex;
+use tokio::task::{JoinError, JoinHandle};
+
+use crate::ScuttleButt;
+
+/// Buffer size for UDP messages.
+const BUF_SIZE: usize = 4096;
+
+/// UDP ScuttleButt server.
+pub struct ScuttleServer {
+    pub scuttlebutt: Arc<Mutex<ScuttleButt>>,
+
+    channel: UnboundedSender<ChannelMessage>,
+    join_handle: JoinHandle<()>,
+}
+
+impl ScuttleServer {
+    /// Launch a new server.
+    ///
+    /// This will start the ScuttleButt server as a new Tokio background task.
+    pub fn spawn(node_id: impl Into<String>, port: u16) -> Self {
+        let (tx, rx) = mpsc::unbounded_channel();
+
+        let scuttlebutt = Arc::new(Mutex::new(ScuttleButt::with_node_id(node_id.into())));
+        let scuttlebutt_server = scuttlebutt.clone();
+
+        let join_handle = tokio::spawn(async move {
+            let _ = listen(rx, scuttlebutt_server, port).await;
+        });
```

Comment:
One of the main issues I see with just running some background tokio server as discussed in our meeting, is that this is lacking transparency to the consumer. If no tokio runtime is present, the function will fail at runtime.

It would of course be possible to leave it up to the consumer to call the server loop's body in a loop, but that then negatively affects the API surface even though people still need to use a runtime anyway. So I've decided to leave it as-is for now.

Reply:
Yes this is ok.

This is a quickwit project crate first, not a general purpose library. (at least for the moment)